src/core/ruby_spy.rs

```diff
@@ -123,6 +165,103 @@ impl RubySpy {
             self.process.pid,
         )
     }
+
+    #[cfg(any(windows, target_os = "linux"))]
+    fn get_native_frames(
+        &mut self,
+        thread: remoteprocess::Thread,
+        lock: remoteprocess::Lock,
+    ) -> Result<Vec<StackFrame>> {
+        let mut native_ips: Vec<u64> = Vec::new();
+        for ip in self.unwinder.cursor(&thread)? {
+            let ip = ip?;
+            native_ips.push(ip)
+        }
+
+        drop(lock);
+
+        let mut native_frames: Vec<StackFrame> = Vec::new();
+        for ip in native_ips {
+            self.symbolicate(ip, &mut |sf| {
+                let name: &str = sf
+                    .function
+                    .as_ref()
+                    .map(|s| s as &str)
+                    .unwrap_or_else(|| "cfunc (unnamed)".into());
+                let name = Name::from(name);
+                let name = name.try_demangle(DemangleOptions::complete());
+
+                native_frames.push(StackFrame {
+                    name: name.into(),
+                    relative_path: sf.filename.clone().unwrap_or_default(),
+                    absolute_path: None,
+                    lineno: sf.line.unwrap_or_default() as u32,
+                });
+            })
+            .unwrap_or_else(|_| {
+                native_frames.push(StackFrame {
+                    name: format!("0x{:x}", ip),
+                    relative_path: "".to_string(),
+                    absolute_path: None,
+                    lineno: 0,
+                });
+            });
+        }
+        Ok(native_frames)
+    }
+
+    fn symbolicate(
+        &mut self,
+        ip: u64,
+        func: &mut dyn FnMut(&remoteprocess::StackFrame),
+    ) -> Result<(), remoteprocess::Error> {
+        match self.symbolicator.symbolicate(ip, true, func) {
+            Ok(_) => Ok(()),
+            Err(e) => {
+                // Do not reload more than once per IP which is not found
```

Comment:
Just curious, is this a performance optimization? Or does it cause an error if we reload more than once per IP?